apps/openmw/mwworld/worldimp.cpp

```diff
@@ -1024,9 +1025,17 @@ namespace MWWorld
 
             float activationDistance = getMaxActivationDistance() + telekinesisRangeBonus;
 
-            facedObject = getFacedObject(activationDistance);
+            facedObject = getFacedObject(activationDistance, distanceToObject, true);
+
+            // Not allowing telekinesis on actors, on doors that teleport to other cells, or on activators
+            // Original engine doesn't allow telekinesis on books or lights, either
+            if (!facedObject.isEmpty() && (facedObject.getClass().isActor()
+                || facedObject.getCellRef().getTeleport() || facedObject.getClass().getTypeName() == "struct ESM::Activator")
```

Comment:
I think it would be better (and consistent with the rest of the code) to use `typeid(ESM::Activator).name()` here. That way if the struct name ever changes, we will get an error at compile time.